File: 2695-find-score-of-an-array-after-marking-all-elements/find-score-of-an-array-after-marking-all-elements.cpp
class Solution {
public:
    long long findScore(vector<int>& nums) {
        long long result = 0;
        priority_queue<vector<int>, vector<vector<int>>, greater<vector<int>>> pq;
        for (int i = 0; i < nums.size(); i++)
        {
            vector<int> temp;
            temp.push_back(nums[i]);
            temp.push_back(i);
            pq.push(temp);
        }
        vector<bool> visited(nums.size(), false);
        while(pq.size() > 0){
            vector<int> temp = pq.top();
            
            pq.pop();
            
            if (!visited[temp[1]])
            {
            result += temp[0];
            visited[temp[1]] = true;
            if (temp[1] > 0)
                visited[temp[1] - 1] = true;
            if (temp[1] < nums.size() -1)
                visited[temp[1]  + 1] = true;
            }
            
        }
        return result;
    }
};

File: 2695-find-score-of-an-array-after-marking-all-elements/find-score-of-an-array-after-marking-all-elements.java
class Solution {
    public long findScore(int[] nums) {
        long result = 0;
        PriorityQueue<Pair<Integer, Integer>> pq = new PriorityQueue<Pair<Integer, Integer>>((p,q) -> Integer.compare(p.getKey() * 10 + p.getValue(), q.getKey() * 10 + q.getValue()));
        for (int i = nums.length - 1; i >=0; i--)
        {
            Pair<Integer, Integer> temp = new Pair<Integer, Integer>(nums[i], i);
            pq.add(temp);
        }
        boolean visited[] = new boolean[nums.length];
        Arrays.fill(visited, false);
        while(pq.size() > 0){
            Pair<Integer, Integer> temp = pq.poll();            
            if (!visited[temp.getValue()])
            {
            result += temp.getKey();
            visited[temp.getValue()] = true;
            if (temp.getValue() > 0)
                visited[temp.getValue() - 1] = true;
            if (temp.getValue() < nums.length -1)
                visited[temp.getValue()  + 1] = true;
            }
            
        }
        return result;
    }
}